### PR TITLE
Implement BackstopJS for visual regression testing

### PR DESCRIPTION
**Pull request by HaSistrunk**
Adds [backstopJS](https://github.com/garris/BackstopJS) for visual regression testing and adds testing reference images that include all the major layouts and page elements for 3 viewport sizes.

Note: When we implement our RAC style library into this site, we'll need to review the scenario selectors and likely update them because of CSS classname changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,18 +30,30 @@ The `build` script bundles four commands into a single bash script
...
+```
 Once the application starts successfully, you should be able to access the application in your browser at `http://localhost:4000`
 
 When you're done, shut down the application with `ctrl+c`.
 
+### Visual Regression Testing
+
+The repository includes [BackstopJS](https://github.com/garris/BackstopJS) to test visual changes to the site by comparing a set of reference images for different screen sizes. Anytime the CSS styles are changed, use BackstopJS to test locally:
+
+1. Build the site and start the application using `bundle exec jekyll serve --skip-initial-build`
```
Comment (helrond) [inline README.md]:
```suggestion
1. Build the site using the steps in the [Local Development section](#local-development) and start the application using `bundle exec jekyll serve --skip-initial-build`
```

**Review by helrond — CHANGES_REQUESTED** [review]
I'm able to get this to run, but I'm consistently getting errors where elements are a couple of pixels off in positioning.

Added one comment so folks build the site using the correct process.

**Comment by HaSistrunk** [comment]
> I'm able to get this to run, but I'm consistently getting errors where elements are a couple of pixels off in positioning.

I couldn't reproduce, but I've updated this to run the tests in Docker, which should hopefully solve that issue. 

Curious what errors, if any, you see now. 

I am seeing periodic but not consistent test fails because of what I believe is a font loading timing issue (comparing the Reference and Tests images that fail show font differences). I haven't found a good solution for this, but I'm wondering if it makes sense to wait and see if this issue persists after implementing the Style Library that imports fonts differently. I know that's not ideal for visual regression testing during Styles implementation, but if the only test fails we see are those fonts, we can work with that for now and troubleshoot later.

**Review by p-galligan — APPROVED** [review]
I don't fully understand all the JS/Backstop code, but I was able to generally follow what was happening.

I managed to build/locally deploy the site and run the tests.

All 15 tests passed for me.

**Comment by HaSistrunk** [comment]
> I don't fully understand all the JS/Backstop code, but I was able to generally follow what was happening.

Most important is understanding that the page snapshots that are taken and compared for visual regression testing are specified in the `backstop.json` file, and then if the `readySelector` or `clickSelector` selectors change in the website UI because of changes to the HTML/CSS, then we'll want to update those selectors so that they are still capturing the page elements correctly.

**Comment by helrond** [comment]
I continue to see small errors in element position. They do not appear to be font-related in the sense that the font appears to be the same size in both screenshots.

We had previously decided not to add visual regression testing to a CI pipeline because it was too inconsistent, so regardless there's going to be some human judgment applied to the visual regression process. I wonder if we've reached the point of diminishing returns in terms of the process, and that we consider this implementation good enough?

**Comment by HaSistrunk** [comment]
> I continue to see small errors in element position. They do not appear to be font-related in the sense that the font appears to be the same size in both screenshots.
> 
> We had previously decided not to add visual regression testing to a CI pipeline because it was too inconsistent, so regardless there's going to be some human judgment applied to the visual regression process. I wonder if we've reached the point of diminishing returns in terms of the process, and that we consider this implementation good enough?

Ok, agreed. And I would say continue to pay attention to BackstopJS updates, because it's pretty actively being developed/improved.

**Pull request merged**